prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -437,60 +418,88 @@ class FHIRRouter(
         reverseFilter: Boolean = false,
         focusResource: Base = bundle
     ): Boolean {
-        val passes = evaluateFilterCondition(
+        val (passes, failingFilterName) = evaluateFilterCondition(
             filters,
             bundle,
             defaultResponse,
             reverseFilter,
             focusResource
         )
         if (!passes) {
-            logFilterResults(filters, bundle, report, receiver, filterType)
+            val filterToLog = "${
+            if ((filterType == ReportStreamFilterType.QUALITY_FILTER && filters === qualityFilterDefault) ||
+                (
+                    filterType == ReportStreamFilterType.PROCESSING_MODE_FILTER &&
+                        filters === processingModeFilterDefault
+                    )
+            )
```

Comment:
The usage of `===` (referential equality operator) is intentional and necessary here because we only want to tag a filter as a default filter if it references the default filter, and not in cases where a filter is equivalent to the default filter but references a filter that was derived from a receiver's settings.

Reply:
This may be worth adding as a comment in the code itself